File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/models/ScheduleEvent.java
package com.neatorobotics.sdk.android.models;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

/**
 * Neato-SDK
 * Created by Marco on 06/05/16.
 * Copyright © 2016 Neato Robotics. All rights reserved.
 */
public class ScheduleEvent implements Serializable {

    private static final String TAG = "ScheduleEvent";

    //public String eventId = "0";
    public String startTime;//24hours format
    public int mode, day;//0 is sunday
    public String boundaryId; //Optional. If provided, the ID of the zone to clean.
    public String boundaryName; //Optional. If provided, the name of the zone to clean.

    public void loadData(JSONObject json) {
        if(json == null) return;
        try {
            if(json.has("startTime") && !json.isNull("startTime")){
                startTime = json.getString("startTime");
            }
            if(json.has("mode") && !json.isNull("mode")){
                mode = json.getInt("mode");
            }else mode = -1;
            if(json.has("day") && !json.isNull("day")){
                day = json.getInt("day");
            }
            if(json.has("boundary")){
                JSONObject boundary = json.getJSONObject("boundary");
                if(boundary != null && boundary.has("id")) {
                    boundaryId = boundary.getString("id");
                }
                if(boundary != null && boundary.has("name")) {
                    boundaryName = boundary.getString("name");
                }
            }
        }catch(JSONException e) {Log.e(TAG, "Exception", e);}
    }
}

File: Neato-SDK/neato-sdk-android/src/main/java/com/neatorobotics/sdk/android/models/ScheduleEvent.kt
/*
 * Copyright (c) 2019.
 * Neato Robotics Inc.
 */

package com.neatorobotics.sdk.android.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
class ScheduleEvent(var startTime: String? = null,//24hours format
                    var mode: CleaningMode = CleaningMode.TURBO,
                    var day: Int = 0,//0 is sunday
                    var boundaryIds: MutableList<String>? = null,
                    var boundaryNames: MutableList<String>? = null,
                    var mapId: String? = null) : Parcelable
